types/mydecimal.go

```diff
@@ -1212,6 +1229,32 @@ func (d *MyDecimal) ToBin(precision, frac int) ([]byte, error) {
 	return bin, err
 }
 
+//	ToHashKey removes the leading zeros and trailing zeros and generates a hash key;
+//	Two Decimals dec0 and dec1 with different fraction will generate the same hash keys if dec0.Compare(dec1) == 0.
+//
+//	For example:
+//		1.1		decimal(2, 1)
+//		1.10	decimal(3, 2)
+//		01.100	decimal(5, 3)
+//	these numbers above will generate a same key.
+func (d *MyDecimal) ToHashKey() ([]byte, error) {
+	_, digitsInt := d.removeLeadingZeros()
+	_, digitsFrac := d.removeTrailingZeros()
+	prec := digitsInt + digitsFrac
+	if prec == 0 { // zeroDecimal
+		prec = 1
+	}
+	buf, err := d.ToBin(prec, digitsFrac)
+	if err == ErrTruncated {
+		// This err is caused by shorter digitsFrac;
+		// After removing the trailing zeros from a Decimal,
+		// 	digitsFrac may be less than the real digitsFrac of the Decimal,
+		// 	thus ErrTruncated may be raised, we can ignore it here.
```

Comment:
@qw4990 Please only leave one space character between "//" and the first word. If possible it's better to let the code comments to be aligned by 80 columns. You can do it in Vim by selecting them in the "VISUAL LINE" mode then press "gq".